app/controllers/books_controller.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+class BooksController < ApplicationController
+  before_action :find_single_book, only: %i[show edit update destroy]
+
+  def index
+    @books = Book.all
+  rescue StandardError => e
+    Rails.logger.error(e.message)
+  end
+
+  def show
+    @reviews = Review.where(book_id: params[:id])
+  rescue StandardError => e
+    Rails.logger.error(e.message)
```

Comment:
It will be better if we can show errors to view, so that it can be better at UX term

Reply:
` def show
    @reviews = Review.where(book_id: params[:id])
    end`
    If i write in this form and error occured, FE will get error by @review.error right?

Reply:
There will not be errors for the query here, but I think New meant to say that instead of rescuing the errors, we can just let them raised and recused only the acceptable ones. 👍

But in case there are errors happened and we would need to send these errors to a user (For example, when they want to create a book, but they didn't specify the name of the book, they must get an descriptive error to enter the name).

In this case, we can use flash (as you already used), but instead of return the normal error text, we can return the error from the record itself.

```ruby
if @book.save
  redirect_to @book
else
  flash[:alert] = @book.errors.full_messages
end
```

Reply:
thx krub :), Now I got it.